bazel/foreign_cc/cel-cpp.patch

```diff
@@ -47,42 +64,22 @@ diff --git a/common/values/value_variant.h b/common/values/value_variant.h
 index 61c19ce5f..fc7969bc8 100644
 --- a/common/values/value_variant.h
 +++ b/common/values/value_variant.h
-@@ -732,6 +732,13 @@
-       const bool rhs_trivial =
-           (rhs.flags_ & ValueFlags::kNonTrivial) == ValueFlags::kNone;
-       if (lhs_trivial && rhs_trivial) {
-+        // We need to suppress the compiler warnings about memory manipulation.
-+        // The memcpy usage here is intentional for performance optimization
-+        // Only suppress this warning on GCC, as Clang doesn't have this warning
-+#if defined(__GNUC__) && !defined(__clang__)
-+#pragma GCC diagnostic push
-+#pragma GCC diagnostic ignored "-Wclass-memaccess"
-+#endif
+@@ -737,9 +737,6 @@
+ #if defined(__GNUC__) && !defined(__clang__)
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wclass-memaccess"
+-#elif defined(__clang__)
+-#pragma clang diagnostic push
+-#pragma clang diagnostic ignored "-Wnontrivial-memcall"
+ #endif
          alignas(ValueVariant) std::byte tmp[sizeof(ValueVariant)];
          // NOLINTNEXTLINE(bugprone-undefined-memory-manipulation)
-         std::memcpy(tmp, std::addressof(lhs), sizeof(ValueVariant));
```

Comment:
Why is this line deleted?

Reply:
I discarded the old patch and iteratively made changes depending on what errors I saw.

For example, unused variable warnings in "TypeId". I remember you mentioned that starting v1.12 we won't be needing some of the things we have in the patch and the baseline I had in mind was to make sure all the triggers pass. 

If you could take a look and point out what other things I should bring back, I will add them back.

Reply:
Oh it's a patch of a patch... It's just hard to parse the difference this way.